bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallHandler.java

```diff
@@ -0,0 +1,179 @@
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TeslaPowerwallHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Paul Smedley - Initial contribution
+ */
+@NonNullByDefault
+public class TeslaPowerwallHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TeslaPowerwallHandler.class);
+
+    private long refreshInterval;
+    private @NonNullByDefault({}) TeslaPowerwallConfiguration config;
+    private String token = "";
+    private @NonNullByDefault({}) TeslaPowerwallWebTargets webTargets;
+    private @Nullable ScheduledFuture<?> pollFuture;
+
+    public TeslaPowerwallHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.warn("This binding is read only");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(TeslaPowerwallConfiguration.class);
+        logger.debug("config.hostname = {}, refresh = {}", config.hostname, config.refresh);
+        if (config.hostname.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Hostname/IP address must be set");
+        } else {
+            webTargets = new TeslaPowerwallWebTargets(config.hostname);
+            refreshInterval = config.refresh;
+            updateStatus(ThingStatus.UNKNOWN);
+            schedulePoll();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        ScheduledFuture<?> pollFuture = this.pollFuture;
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        logger.debug("Scheduling poll for 1 second out, then every {} s", refreshInterval);
+        this.pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 1, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollStatus();
+        } catch (IOException e) {
+            logger.debug("Could not connect to Tesla Powerwall", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (RuntimeException e) {
+            logger.warn("Unexpected error connecting to Tesla Powerwall", e);
```

Comment:
```suggestion
```
As the framework logs already the same info in the next line (updateStatu) this can be removed.